### PR TITLE
Add smhasher

### PR DESCRIPTION
**Pull request by mvdbeek**
* [x] I have read the guidelines above.
* [x] This PR adds a new recipe.
* [ ] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

### CODE REVIEW

File: recipes/smhasher/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  url: https://pypi.python.org/packages/5f/b2/421c060300f1449646787273ecd7df558f6f8e4caee3d6fbe8b193cd60d4/smhasher-0.150.1.tar.gz
+  md5: 4fc61f36b141e137dd2d71ecd0167866
+
+requirements:
+  build:
+    - python
+
+  run:
+    - python
+    - libgcc
```
Comment (bgruening) [inline recipes/smhasher/meta.yaml]: @mvdbeek this is only needed if you compile something afaik, do you really need it?

Comment (bgruening) [inline recipes/smhasher/meta.yaml]: Yes there is. Conda-forge is still using the build-in compiler.

Comment (mvdbeek) [inline recipes/smhasher/meta.yaml]: Ahh, that explains it. So bets practice is to include gcc then, right?

Comment (bgruening) [inline recipes/smhasher/meta.yaml]: Yes this is bioconda best practice at the moment.

File: recipes/smhasher/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+requirements:
+  build:
+    - python
+    - gcc [linux]
+    - llvm [osx]
+
+  run:
+    - python
+    - libgcc
```
Comment (bgruening) [inline recipes/smhasher/meta.yaml]: libgcc only on linux see here: https://github.com/bioconda/bioconda-recipes/blob/master/GUIDELINES.md#cc

**Pull request merged**